streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -28,7 +28,7 @@
     private final Predicate<K, V> predicate;
     private final boolean filterNot;
 
-    private boolean sendOldValues = false;
+    private boolean sendOldValues = true;
```

Comment:
Samza does not have a high-level DSL interface, so it dos not have this issue.

This issue is related to the more general question that when should we materialize a KTable object with a state store, and currently the rule is that stateless operators like filters should not necessarily materialize it. That being said, if a further downstream operator is stateful, it will cause backward propagation of setting `enableSendOldValues` to true up till the source KTable, as I mentioned in the email thread.